lib/inspec/resources/users.rb

```diff
@@ -425,10 +463,26 @@ def credentials(username)
         multiple_values: false
       ).params
 
+      cmd = inspec.command("echo $((($(date +%s) - $(date -d '#{params["Last password change"]}' '+%s'))/86400))")
```

Comment:
Do we need to do math in a `inspec.command`?

Reply:
hi miah, i just pulled this request because no one answered to my "discussion" Topic :)
#4432.
I opened a bug and described why i did math calculation on the target. The reason for doing it on the target is the localization. Each target could have a different localization and therefor ist close to impossible to calculate it by ruby. Next to this the ruby "date" class is not available from inside inspec. Same issue as for some AWS ressources.

I am totally open minded at this Topic. Any discussion and ideas are welcome and should be documented issue #4432.

Reply:
"localization" is the reason why you subtract and divide by 86400 on the client?

Reply:
There are 3 reason for calculation. 
1) Localization of the client 
2) harmonize output to windows. 
3) Ruby date class not a part of "inspec" package.

Reply:
Hi @mhackethal ,
  [`date`](https://ruby-doc.org/stdlib-2.6.4/libdoc/date/rdoc/index.html), is the standard Ruby library that includes the `Date` class, and as such, it is included with every Ruby installation. You can (and should) do any conversion in Ruby.

Reply:
I've got to push back on this one.

1. Too complicated.
2. `date -d` doesn't do what you think it does (on all platforms).
3.  Ruby can do all the conversions for you:

```ruby
>> require "date"
=> true
>> d0 = Date.parse "2019-01-01"
=> #<Date: 2019-01-01 ((2458485j,0s,0n),+0s,2299161j)>
>> Date.today - d0
=> (266/1)
```

Reply:
i am sorry. Youre totally right. Looks like i did something wrong on my end. (big sorry)
On my first tests i received an error when i used require "date" so i thought its not usable.
But looks like i did something wrong - its working now.
```

      #cmd = inspec.command("echo $((($(date +%s) - $(date -d '#{params["Last password change"]}' '+%s'))/86400))")
      #dayslastset = convert_to_i(cmd.stdout.chomp) if cmd.exit_status == 0
      dparse = Date.parse "#{params['Last password change']}"
      dayslastset = (Date.today - dparse).to_i
```
I changed the the code and tested on:
RHEL 6 , 7
Ubuntu 12.x,14,x,16,x
Debian 9.x
Debian 10.x

```
You are currently running on:
    Name:      redhat
    Families:  redhat, linux, unix, os
    Release:   6.10
    Arch:      x86_64
inspec> describe user("root") do
inspec>   its('passwordage') { should eq 0 }
inspec> end

Profile: inspec-shell
Version: (not specified)
  User root
     [FAIL]  passwordage should eq 0
     expected: 0
          got: 9
     (compared using ==)
Test Summary: 0 successful, 1 failure, 0 skipped
```
```
You are currently running on:
    Name:      ubuntu
    Families:  debian, linux, unix, os
    Release:   16.04
    Arch:      x86_64
inspec> describe user("root") do
inspec>   its('passwordage') { should eq 0 }
inspec> end
Profile: inspec-shell
Version: (not specified)
  User root
     [FAIL]  passwordage should eq 0
     expected: 0
          got: 76
     (compared using ==)
Test Summary: 0 successful, 1 failure, 0 skipped
```

I tested both ways onetime with the "cmd" parser and one time with the ruby parser. On my systems the result was equal on both.
I personally would prefer the "cmd" way because i have the following situation in my mind:
- InSpec uses : dd/mm/yyyy
- Target uses : mm/dd/yyyy
Same for yyyy/mm/dd vs. yyyy/dd/mm

Any hint on which distro "date -d" will not do what i expect it to do? (just for personal intrest on this topic).

I will push an update to the code later. Thank you for your time and supporting me on this topic!

Reply:
All done and updated.
Does anyone can give me some suggestion about CodeClimat and the failing BK Test for Windows?
Is this something i should solve?

Reply:
@miah any feedback for @mhackethal on this comment?

Reply:
I've reviewed the codeclimate complaints here and they are valid, but don't need to be addressed by this PR.

Reply:
Can i consider this as no More Tasks for me?

Reply:
Depends on the other reviewers =)